gravitee-am-service/src/main/java/io/gravitee/am/service/impl/ApplicationServiceImpl.java

```diff
@@ -244,6 +242,22 @@
         oAuthSettings.setTokenEndpointAuthMethod(ClientAuthenticationMethod.CLIENT_SECRET_BASIC);
         oAuthSettings.setRedirectUris(newApplication.getRedirectUris());
         applicationSettings.setOauth(oAuthSettings);
+
+        // apply default SAML 2.0 settings
+        if (ApplicationType.SERVICE != application.getType()) {
+            if (!ObjectUtils.isEmpty(newApplication.getRedirectUris())) {
+                try {
+                    final String url = newApplication.getRedirectUris().get(0);
+                    ApplicationSAMLSettings samlSettings = new ApplicationSAMLSettings();
+                    samlSettings.setEntityId(UriBuilder.fromHttpUrl(url).buildRootUrl());
+                    samlSettings.setAttributeConsumeServiceUrl(url);
+                    applicationSettings.setSaml(samlSettings);
+                } catch (Exception ex) {
+                    // silent exception
+                    // redirect_uri can use custom URI (especially for mobile deep links)
```

Comment:
Maybe having a `log.debug` for further help ?

Reply:
FP